Shopify/cli
# Use app context in functions

### WHY are these changes introduced?
Start using `linkedAppContext` in some commands.



### WHAT is this pull request doing?
- Modifies all function commands to use the generic `linkedAppContext` instead of `ensureConnectedAppFunctionContext`
- Create new type `AppLinkedInterface` for better DX
- Update some names and types
- Delete unnecessary tests and functions related to `ensureConnectedAppFunctionContext` 🚀 



### How to test your changes?
- Run the function commands, here are some scenarios:
  - Logged out
  - In a new app
  - With an old toml (just a toml with `client_id = 2222`)
 
Try to run the commands in those scenarios and then repeat, the second time everything should work automatically.


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
